crypto/getenv.c

```diff
@@ -13,9 +13,45 @@
 
 #include <stdlib.h>
 #include "internal/cryptlib.h"
+#include "e_os.h"
 
 char *ossl_safe_getenv(const char *name)
 {
+#if (defined(OPENSSL_SYS_WINDOWS)) && defined(CP_UTF8) && !defined(_WIN32_WCE)
+
+    if (GetEnvironmentVariableW(L"OPENSSL_WIN32_UTF8", NULL, 0) == 0)
+        return getenv(name);
+
+    /* convert name to wide string */
+    int rsize = mbstowcs(NULL, name, 0) + 1;
+    WCHAR * namew = _alloca(rsize * sizeof(WCHAR));
```

Comment:
not sure when _malloca got introduced.. but it is a securer version.

Reply:
mbstowcs_s is only better when programmers assume buffer size, for example:

char [255] buffer;

My code is equivalent: query for correct buffer len, then use it.